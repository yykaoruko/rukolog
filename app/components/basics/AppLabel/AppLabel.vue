<template>
  <div class="app-label">
    {{ label }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
  name: 'AppLabel',
  props: {
    label: {
      type: String,
      required: true,
      validator(value) {
        return value !== '';
      },
    },
  },
});
</script>

<style lang="scss" scoped>
.app-label {
  color: inherit;
}
</style>
