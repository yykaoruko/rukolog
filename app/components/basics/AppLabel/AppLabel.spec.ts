import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import AppLabel from './AppLabel.vue';

describe('AppLabel', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(AppLabel, {
        propsData: {
          label: 'label',
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
