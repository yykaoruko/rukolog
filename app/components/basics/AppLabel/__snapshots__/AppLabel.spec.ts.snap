// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLabel mount snapshot 1`] = `
<div
  class="app-label"
>
  
  label

</div>
`;
